### PR TITLE
Cockpit 214

### PR DESCRIPTION
**Pull request by martinpitt**
Rendered: https://martinpitt.github.io/cockpit-project.github.io/blog/cockpit-214.html

### CODE REVIEW

File: _posts/2020-03-04-cockpit-214.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+tags: cockpit linux
+slug: cockpit-214
+category: release
+summary: Cockpit with bug fixes
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. Here are the release notes from version 214.
+
+### Networking: Improve Firewall page in unprivileged mode
```
Comment (garrett) [inline _posts/2020-03-04-cockpit-214.md]:
```suggestion
### Networking: List Firewall active zones when unprivileged
```

Comment (garrett) [inline _posts/2020-03-04-cockpit-214.md]:
```suggestion
### Networking: List Firewall active zones when unprivileged
```

File: _posts/2020-03-04-cockpit-214.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+[firewalld](https://firewalld.org/) recently became more restrictive for non-superuser mode: It is now not possible any more to list services in a zone.  As a result, the Firewall page was completely empty in unprivileged mode.
+
+With this Cockpit version, the page at least shows the active zones again.
```
Comment (garrett) [inline _posts/2020-03-04-cockpit-214.md]:
```suggestion
[firewalld](https://firewalld.org/) recently became more restrictive. Listing services in a zone is no longer possible for anyone but superusers. As a result, the Firewall page was empty in unprivileged mode.

With this new Cockpit version, the Firewall page shows active zones again, however without services listed.
```

File: _posts/2020-03-04-cockpit-214.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Selenium tests deprecation
+
+We started to adjust many of our "verify" browser tests to be non-destructive, so that they can all run on an already existing test VM without needing to reset it. This was the primary remaining reason for still having the selenium test suite.
+
+The selenium tests have much less coverage and are much harder to debug than the verify tests. Thus we will not develop them any further, and will eventually remove them. If you work with them, please get in touch with us.
```
Comment (garrett) [inline _posts/2020-03-04-cockpit-214.md]:
```suggestion
The Cockpit team has adjusted many "verify" browser tests to be non-destructive. This reworking was done to allow test VMs to be reused without resetting, which was our primary remaining reason to use Selenium tests.

As Selenium are much harder to debug than verify tests and coverage is not quite as complete, we will not develop them further. Eventually, the Selenium tests will be completely removed.

If you work with Selenium tests, please get in touch with us.
```

**Comment by martinpitt** [comment]
Thanks @garrett! Applied, with a few tweaks to point out that the properties of our selenium tests are due to our concrete tests and their framework, not necessarily because Selenium is bad in general.

File: _posts/2020-03-04-cockpit-214.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+[firewalld](https://firewalld.org/) recently became more restrictive. Listing services in a zone is no longer possible for anyone but superusers. As a result, the Firewall page was empty in unprivileged mode.
+
+With this new Cockpit version, the Firewall page shows active zones again, however without services listed.
+
+### Selenium tests deprecation
+
+The Cockpit team has adjusted many "verify" browser tests to be non-destructive. This reworking was done to allow test VMs to be reused without resetting, which was our primary remaining reason for still having the Selenium tests.
+
+As the Selenium are much harder to debug than verify tests and coverage is not quite as complete, we will not develop them further. Eventually, the Selenium tests will be completely removed.
```
Comment (garrett) [inline _posts/2020-03-04-cockpit-214.md]:
```suggestion
Compared to verify tests, Selenium tests are much harder to debug and coverage is not quite as complete. Therefore, we will no longer develop Selenium tests. Eventually, the Selenium tests will be completely removed.
```

Missing the word "tests" after Selenium and the flow was a bit clunky still. Revised to include both. Additionally, I specified Selenium once more as it was not clear which tests will be abandoned. Clarity is more important than redundancy. :wink:

**Pull request merged**